compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -653,6 +652,17 @@ fn encode_ty<'tcx>(
             compress(dict, DictKey::Ty(ty, TyQ::None), &mut s);
             typeid.push_str(&s);
         }
+        ty::CoroutineClosure(def_id, args) => {
+            // u<length><name>[I<element-type1..element-typeN>E], where <element-type> is <subst>,
+            // as vendor extended type.
+            let mut s = String::new();
+            let name = encode_ty_name(tcx, *def_id);
+            let _ = write!(s, "u{}{}", name.len(), &name);
+            let parent_args = tcx.mk_args(args.as_coroutine_closure().parent_args());
+            s.push_str(&encode_args(tcx, parent_args, dict, options));
+            compress(dict, DictKey::Ty(ty, TyQ::None), &mut s);
+            typeid.push_str(&s);
+        }
 
         ty::Coroutine(def_id, args, ..) => {
```

Comment:
For coroutines, we should actually also encode `args.as_closure().kind_ty()` in addition to the parent args. But if you don't want to do this, I could craft a test where this matters.

Reply:
I don't object to doing it, but would prefer it in a follow up to keep things moving.

The other thing to note here is that lacking the `kind_ty` won't break anything, it will just make our alias sets not-as-good-as-they-could-be, since coroutines of different kinds will be swappable by an attacker.

Reply:
Yeah, making them distinguishable would be useful though.